### ISSUE TITLE
Licence of sly-repl-ansi-color

### ISSUE THREAD

**Issue by brandelune** [PuercoPop/sly-repl-ansi-color#7]
Hi,

In a thread in the emacs-devel maillist, the licensing situation for emacs packages provided through Emacs package archives has been under focus. I have volunteered to contact the authors of packages that have a license that is incompatible with Emacs, which is now under GPL-3+.

See https://lists.gnu.org/archive/html/emacs-devel/2017-07/msg01069.html

So I wonder if you could consider to change the license of your package to GPL-3+?

Yours,

Jean-Christophe Helary

**Comment by PuercoPop** [comment]
Hi, I'm not sure I follow. This package is licensed under GPLv2+. Why would that be incompatible with Emacs?

**Comment by brandelune** [comment]
Thank you very much for your reply.

On the project page, the licence is given as GPL-2.0.
The License part of the readme does not indicate clearly that the license is GPL-2.0+ and the LICENSE file does not specify that clearly either.

Would you mind adding "and any later version" to the License part of the readme ?

Thank you in advance.

Jean-Christophe

**Comment by mfiano** [comment]
This project has been removed from MELPA because it is not compatible with GPLv3.

https://github.com/melpa/melpa/commit/c3366117f3399660509f62849a88dc8be56a1fef

**Comment by PuercoPop** [comment]
@mfiano it is now hosted in the @emacs-pe elpa.

**Comment by joaotavora** [comment]
Javier, do you have anything against doing what Jean-Christophe proposes,
i.e. more explicitly state that the license is actually GPLv2+? It could
then be included in the two repos.

**Comment by mfiano** [comment]
I would really like to see this in MELPA. Adding yet another package archive to all of my Emacs machines is not ideal just for 1 package.

**Comment by PuercoPop** [comment]
@joaotavora No, in fact I was surprised that it isn't GPLv2+. I intend[ed] to do when got blindsided by a deadline I was unaware of. Or a link as to why they think GPLv2 is unacceptable. I'll probably get around to addressing this issue this this weekend.

**Comment by brandelune** [comment]
It is not that GPLv2 is unacceptable, it is that GLPv2 (only) is not compatible with GLP3. To be compatible you need to change to GPL2+ so that distribution can also be made under GPL3. Thank you in advance !

**Comment by PuercoPop** [comment]
@brandelune I understand the claim, I meant haven't seen the link to an argument as to why. And also are they treating elisp code as 'derivative works'? From what happened with CLISP I can't say I agree with the FSF's interpretation of derivative work. (But that is not relevant to addressing the issue)

**Comment by PuercoPop** [comment]
@brandelune Could you give me a hand, here do I have to add the or Later phrase in the License? 🙏

**Comment by PuercoPop** [comment]
[Don't this](https://github.com/PuercoPop/sly-repl-ansi-color/blob/master/LICENSE#L298-L299) mean the package is GPLv2+?

**Comment by brandelune** [comment]
https://www.gnu.org/licenses/gpl-faq.en.html#v2v3Compatibility

**Comment by brandelune** [comment]
It is not clear that part of the text applies to your code. I'll send you a pull request later if you don't mind.  Thank you very much for your help.

**Comment by psilord** [comment]
Any news on this topic? I'd like to use this package, but it seems still not to be in melpa. Was the license ever changed or updated so the tools would pass the package?

**Comment by PuercoPop** [comment]
@psilord AFAIU, This package was GPLv2+ from the start, which is a 'superspet of GPLv3+. So I'm not clear why it was removed from MELPA. If that is not the case I'll be more than happy to accept any pull request clarifying the license.

**Comment by brandelune** [comment]
Apologies. I ment to sent a pull request to clarify the license. I'll do that later.

Jean-Christophe

**Comment by mfiano** [comment]
Yeah I thought so too. So confused why this package was erroneously removed from MELPA:

https://github.com/PuercoPop/sly-repl-ansi-color/blob/master/LICENSE#L299

**Comment by tarsius** [comment]
> Yeah I thought so too. So confused why this package was erroneously removed from MELPA:
>
> https://github.com/PuercoPop/sly-repl-ansi-color/blob/master/LICENSE#L299

By adding that LICENSE file this package was released under exactly version two of the GPL. The part of that file you are linking to comes after this:

```
[actual license text]

                     END OF TERMS AND CONDITIONS

            How to Apply These Terms to Your New Programs

....

[you link somewhere here]
```

You have to actually follow those instructions for the permission statement to actually apply to the program.
